src/progress/progress_set.rs

```diff
@@ -421,20 +427,27 @@ impl ProgressSet {
     /// Doing this will set the `recent_active` of each peer to false.
     ///
     /// This should only be called by the leader.
-    pub fn quorum_recently_active(&mut self, perspective_of: u64) -> bool {
+    pub fn quorum_recently_active<F>(&mut self, perspective_of: u64, mut f: F) -> bool
+    where
+        F: FnMut(u64, bool),
+    {
         let mut active = HashSet::default();
-        for (&id, pr) in self.voters_mut() {
+        let learners = self.configuration.take_learners();
+        for (&id, pr) in self.iter_mut() {
```

Comment:
So if the peer is a learner, it will be inserted into `active`? I think it's wrong.